### PR TITLE
Primary key for column schema builder

### PR DESCRIPTION
**Pull request by mdmunir**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | -
| Fixed issues  | #11638 #10889

* Introduce more relevant name for auto increment column
```php
[
    'id' => $this->serial(), // instead of $this->primaryKey()
]
```
* Add `primaryKey()` to `ColumnSchemaBuilder`
```php
        $this->createTable($cache->cacheTable, [
            'id' => $this->string(128)->primaryKey(),
            'expire' => $this->integer(),
            'data' => $this->binary(),
        ], $tableOptions);
```

### CODE REVIEW

File: framework/db/ColumnSchemaBuilder.php [hunk 1]
```diff
@@ -317,12 +332,13 @@ protected function buildNotNullString()
...
-     * Builds the unique constraint for the column.
-     * @return string returns string 'UNIQUE' if [[isUnique]] is true, otherwise it returns an empty string.
+     * Builds the unique or primary constraint for the column.
+     * @return string returns string 'PRIMARY KEY' if [[isPrimary]] is true, else
+     * returns string 'UNIQUE' if [[isUnique]] is true, otherwise it returns an empty string.
      */
     protected function buildUniqueString()
     {
-        return $this->isUnique ? ' UNIQUE' : '';
+        return $this->isPrimary ? ' PRIMARY KEY' : ($this->isUnique ? ' UNIQUE' : '');
```
Comment (dynasource) [inline framework/db/ColumnSchemaBuilder.php]: could be improved on readability. Switch statement on true.

**Comment by dynasource** [comment]
to add to the conversion:
https://dev.mysql.com/doc/refman/5.5/en/numeric-type-overview.html
'SERIAL is an alias for BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE.'

File: framework/db/SchemaBuilderTrait.php [hunk 2]
```diff
@@ -44,9 +44,9 @@
      * @param integer $length column size or precision definition.
      * This parameter will be ignored if not supported by the DBMS.
      * @return ColumnSchemaBuilder the column instance which can be further customized.
-     * @since 2.0.6
+     * @since 2.0.11
      */
-    public function primaryKey($length = null)
+    public function serial($length = null)
     {
         return $this->getDb()->getSchema()->createColumnSchemaBuilder(Schema::TYPE_PK, $length);
```
Comment (dynasource) [inline framework/db/SchemaBuilderTrait.php]: if you look at mysqls definition for SERIAL (SERIAL is an alias for BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE.), this is a bit confusing and perhabs error prone, using the Schema::TYPE_PK.

**Review by dynasource — CHANGES_REQUESTED** [review]
I am missing some tests for other DBs like postgresql & mssql. Is there a specific reason no to include?

File: framework/db/ColumnSchemaBuilder.php (hunk 1 above)
Comment (mdmunir) [inline framework/db/ColumnSchemaBuilder.php]: ya.. you are right.

File: framework/db/SchemaBuilderTrait.php (hunk 2 above)
Comment (mdmunir) [inline framework/db/SchemaBuilderTrait.php]: only mysql right? IMO, `serial()` still more make sense than `primaryKey()`.

Comment (dynasource) [inline framework/db/SchemaBuilderTrait.php]: please elaborate

Comment (rob006) [inline framework/db/SchemaBuilderTrait.php]: Why not `autoIncrement()`?

File: framework/db/ColumnSchemaBuilder.php [hunk 3]
```diff
@@ -154,6 +159,16 @@ public function null()
     }
 
     /**
+     * Adds a `PRIMARY KEY` constraint to the column.
+     * @return $this
```
Comment (SilverFire) [inline framework/db/ColumnSchemaBuilder.php]: `@since`

File: framework/db/SchemaBuilderTrait.php [hunk 4]
```diff
@@ -270,4 +270,22 @@ public function money($precision = null, $scale = null)
         }
         return $this->getDb()->getSchema()->createColumnSchemaBuilder(Schema::TYPE_MONEY, $length);
     }
+
+    /**
+     * @deprecated since version 2.0.11
+     * @see [[serial()]]
```
Comment (SilverFire) [inline framework/db/SchemaBuilderTrait.php]: `@since 2.0.6`

File: framework/db/SchemaBuilderTrait.php [hunk 5]
```diff
@@ -270,4 +270,22 @@ public function money($precision = null, $scale = null)
...
+     */
+    public function primaryKey($length = null)
+    {
+        return $this->serial($length);
+    }
+
+    /**
+     * @deprecated since version 2.0.11
+     * @see [[bigSerial()]]
```
Comment (SilverFire) [inline framework/db/SchemaBuilderTrait.php]: `@since 2.0.6`

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Comment by SilverFire** [comment]
I don't like `serial()` much. This name is not self-explanatory and does not reference to the appropriate DBMS concept.

**Comment by mdmunir** [comment]
> I don't like serial() much

postgre use `serial` as type.

File: framework/db/ColumnSchemaBuilder.php (hunk 3 above)
Comment (SilverFire) [inline framework/db/ColumnSchemaBuilder.php]: Resolved

File: framework/db/SchemaBuilderTrait.php [hunk 6]
```diff
@@ -270,4 +270,24 @@ public function money($precision = null, $scale = null)
         }
         return $this->getDb()->getSchema()->createColumnSchemaBuilder(Schema::TYPE_MONEY, $length);
     }
+
+    /**
+     * @deprecated since version 2.0.11
```
Comment (SilverFire) [inline framework/db/SchemaBuilderTrait.php]: Deprecation does not mean that the method description should be dropped. Could you restore it?

**Review by SilverFire — CHANGES_REQUESTED** [review]
DOCS update is required

**Comment by samdark** [comment]
Won't make it into 2.0.

**Pull request closed without merging**